### DISCUSSION TITLE
Interaction Helmchart interval and HelmRepository interval

### DISCUSSION THREAD

**Discussion by dejoost** [fluxcd/flux2#2716] [Q&A] [answered]
After reviewing the documentation it is not 100% clear to me how the interaction happens between HelmChart and HelmRepository.

If the HelmRepository has an interval of e.g. 10m and the HelmChart referencing this repo an interval of 1m, will charts be updated faster then 10m on changes, or will it check 10 times the same HelmRepository artifact before the SourceController will do a next check and updates the artifact. A bit differently stated, does it make sense to put the interval value on the HelmChart lower than the interval on the HelmRepository?

**Comment by hiddeco** [comment, marked as answer]
Given the update of a HelmRepository triggers the update of a HelmChart, in most cases, it does not. The value is mostly set to e.g. ensure the artifact continues to exist in storage.

**Reply by dejoost** [reply]
interesting, does it mean, if the artifact is cleaned up, it will be recreated, or it updates a timestamp on the artifact in order for the cleanup to skip the artifact?

And you said, "in some cases", are there any specific cases where it could interact with something else besides the helmrelease object or its artifacts?

**Reply by kingdonb** [reply]
> if the artifact is cleaned up, it will be recreated, or it updates a timestamp on the artifact in order for the cleanup to skip the artifact?

Yes

> are there any specific cases where it could interact with something else besides the helmrelease object or its artifacts?

No, not at this time. It's possible this could change in the future, one other possibility is that the artifact has changed on the Helm Repo itself. (That wouldn't usually be the case since Helm artifacts are meant to be immutable at a given version.)
